### PR TITLE
Extractor: added some extractors

### PR DESCRIPTION
**Pull request by keyiflerolsun**
Odnoklassniki,
TauVideo,
SibNet,
ContentX,
Hotlinger,
HDMomPlayer,
HDPlayerSystem,
VideoSeyred,
PeaceMakerst,
HDStreamAble,
RapidVid,
TRsTX,
VidMoxy,
PixelDrain,
MailRu

All of runs on my repo : https://github.com/keyiflerolsun/Kekik-cloudstream

### CODE REVIEW

**Review by fire-light42 — COMMENTED** [review]
1. You use `Log.d("Kekik_${this.name}", message)` everywhere, why? You can do a simple extention function on ExtractorApi called `ExtractorApi.logKekik(message:String)`
2. snake_case everywhere? camelCase is what kotlin standard is
3. You should use the new INFER_TYPE that figures out the link type based on the link instead of legacy ism3u8
4. you can use the :? throw/return operator to discard null values instead of doing if (x != null) { rest of code }

Very nice addition! 🥳 
I hope these small changes for code quality can be made before merge, however if you simply don't have the time then I can merge.

File: app/src/main/java/com/lagradost/cloudstream3/extractors/ContentXExtractor.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        val i_dublaj = Regex(""",\"([^']+)\",\"Türkçe""").find(i_source)!!.groups[1]?.value
+        if (i_dublaj != null) {
+            val dublaj_source  = app.get("https://contentx.me/source2.php?v=${i_dublaj}", referer=ext_ref).text
+            val dublaj_extract = Regex("""file\":\"([^\"]+)""").find(dublaj_source)!!.groups[1]?.value ?: throw ErrorLoadingException("dublaj_extract is null")
+            val dublaj_link    = dublaj_extract.replace("\\", "")
+
+            callback.invoke(
+                ExtractorLink(
+                    source  = "${this.name} Türkçe Dublaj",
```
[the remark this answers is not in the document]
Comment (keyiflerolsun) [reply app/src/main/java/com/lagradost/cloudstream3/extractors/ContentXExtractor.kt:44]: In this resource, native language comes by default. Secondly, there are dubbing options, if any.

File: app/src/main/java/com/lagradost/cloudstream3/extractors/ContentXExtractor.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                source  = this.name,
+                name    = this.name,
+                url     = m3u_link,
+                referer = url,
+                quality = Qualities.Unknown.value,
+                isM3u8  = true
+            )
+        )
+
+        val i_dublaj = Regex(""",\"([^']+)\",\"Türkçe""").find(i_source)!!.groups[1]?.value
```
[the remark this answers is not in the document]
Comment (keyiflerolsun) [reply app/src/main/java/com/lagradost/cloudstream3/extractors/ContentXExtractor.kt:36]: In this resource, native language comes by default. Secondly, there are dubbing options, if any.

**Comment by keyiflerolsun** [comment]
I can answer your other questions by saying "I am a Python software developer" :) I liked your application very much and I use it actively.  This is all the support I could get :)

**Review by fire-light42 — APPROVED** [review]
This can be merged if you want 👍

**Comment by keyiflerolsun** [comment]
> This can be merged if you want 👍

I'm doing the final checks. "INFER_TYPE" is checked with endswith, but in some providers it needs to be checked with contains because time-limited links can be created with different parameters.

**Comment by fire-light42** [comment]
No, it should only do the path not params, https://github.com/recloudstream/cloudstream/blob/3d90af29eb055cf8d69ce44820f811cfd9ca010e/app/src/main/java/com/lagradost/cloudstream3/utils/ExtractorApi.kt#L281

but why did you close the pr?
```kotlin
import java.net.URL
fun main() {
    val ret = URL("https://hello/world.m3u8?params=123&otherparam=234") 
    println("Path: ${ret.path}") // prints "/world.m3u8"
}
```

**Comment by fire-light42** [comment]
If you feel like INFER_TYPE is somehow invalid then you are free to update it, the plan is to make it more robust than regular providers without having to write .contains everywhere

**Comment by keyiflerolsun** [comment]
I need to do some more testing locally before merging :) i will come back 🕊

**Pull request closed without merging**